homeassistant/components/plugwise/gateway.py

```diff
@@ -64,95 +66,99 @@ async def async_setup_entry_gw(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     try:
         connected = await api.connect()
-
         if not connected:
-            _LOGGER.error("Unable to connect to Smile")
+            _LOGGER.error("Unable to connect to the Smile/Stretch")
             raise ConfigEntryNotReady
-
     except InvalidAuthentication:
         _LOGGER.error("Invalid username or Smile ID")
         return False
-
     except PlugwiseException as err:
-        _LOGGER.error("Error while communicating to device %s", api.smile_name)
+        _LOGGER.error("Error while communicating to the Smile/Stretch")
         raise ConfigEntryNotReady from err
-
     except asyncio.TimeoutError as err:
-        _LOGGER.error("Timeout while connecting to Smile %s", api.smile_name)
+        _LOGGER.error("Timeout while connecting to the Smile/Stretch")
         raise ConfigEntryNotReady from err
 
+    # Migrate to a valid unique_id when needed
+    if entry.unique_id is None:
+        if api.smile_version[0] != "1.8.0":
+            hass.config_entries.async_update_entry(entry, unique_id=api.smile_hostname)
+
     update_interval = timedelta(
         seconds=entry.options.get(
             CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL[api.smile_type]
         )
     )
 
-    async def async_update_data():
+    async def async_update_gw_data():
         """Update data via API endpoint."""
+        _LOGGER.debug("Updating %s", api.smile_name)
         try:
-            async with async_timeout.timeout(DEFAULT_TIMEOUT):
-                await api.full_update_device()
+            async with async_timeout.timeout(update_interval.seconds):
+                await api.update_gw_devices()
                 return True
         except XMLDataMissingError as err:
             raise UpdateFailed("Smile update failed") from err
+        except PlugwiseException as err:
+            _LOGGER.debug("Updating failed, generic failure for %s", api.smile_name)
```

Comment:
This debug message isn't needed, instead, add the smile name to the UpdateFailed raise, as that will already log it.